gateway-service/src/test/java/org/zowe/apiml/gateway/controllers/GatewayExceptionHandlerTest.java

```diff
@@ -33,75 +38,111 @@
-        );
-    }
+        @ParameterizedTest
+        @CsvSource({
+            "400,org.zowe.apiml.common.badRequest",
+            "401,org.zowe.apiml.common.unauthorized",
+            "403,org.zowe.apiml.security.forbidden",
+            "404,org.zowe.apiml.common.notFound",
+            "405,org.zowe.apiml.common.methodNotAllowed",
+            "415,org.zowe.apiml.common.unsupportedMediaType",
+            "500,org.zowe.apiml.common.internalServerError",
+            "503,org.zowe.apiml.common.serviceUnavailable",
+        })
+        void givenErrorResponse_whenCallGateway_thenDecorateIt(int code, String messageKey) throws MalformedURLException {
+            mockException.set(WebClientResponseException.create(code, "msg", null, null, null));
+
+            given().when()
+                .get(new URL(basePath + "/serv1ce/api/v1/test"))
+                .then()
+                .statusCode(code)
+                .body("messages[0].messageKey", containsString(messageKey));
+        }
 
+        Stream<Arguments> getExceptions() {
+            return Stream.of(
+                Arguments.of(new ForbidSlashException(""), 400, "org.zowe.apiml.gateway.requestContainEncodedSlash"),
+                Arguments.of(new ForbidCharacterException(""), 400, "org.zowe.apiml.gateway.requestContainEncodedCharacter"),
+                Arguments.of(new ResponseStatusException(HttpStatusCode.valueOf(504)), 504, "org.zowe.apiml.gateway.responseStatusError")
+            );
+        }
 
-    @ParameterizedTest
-    @MethodSource("getExceptions")
-    void whenExceptionIsThrown_thenTranslateToCorrectMessage(Exception ex, int status, String message) throws MalformedURLException {
-        mockException.set(ex);
-        given().when()
-            .get(new URL(basePath + "/serv1ce/api/v1/test"))
-            .then()
-            .statusCode(status)
-            .body("messages[0].messageKey", containsString(message));
-    }
+        @ParameterizedTest
+        @MethodSource("getExceptions")
+        void whenExceptionIsThrown_thenTranslateToCorrectMessage(Exception ex, int status, String message) throws MalformedURLException {
+            mockException.set(ex);
+            given().when()
+                .get(new URL(basePath + "/serv1ce/api/v1/test"))
+                .then()
+                .statusCode(status)
+                .body("messages[0].messageKey", containsString(message));
+        }
+
+        @Configuration
+        @Profile("gatewayExceptionHandlerTest")
+        static class Config {
+
+            @Bean
+            GlobalFilter exceptionFilter() {
+                return (exchange, chain) -> Mono.error(mockException.get());
+            }
 
-    @Configuration
-    @Profile("gatewayExceptionHandlerTest")
-    static class Config {
+        }
+
+    }
 
-        @Bean
-        GlobalFilter exceptionFilter() {
-            return (exchange, chain) -> Mono.error(mockException.get());
+    @Nested
+    @SpringBootTest
+    class Methods {
```

Comment:
maybe some better name? And does it need to be a different Springboo test than the parent test class?